src/PAModel/CanvasDocument.cs

```diff
@@ -584,7 +584,15 @@ internal void StabilizeAssetFilePaths(ErrorContainer errors)
                 var withoutPrefix = GetAssetFilePathWithoutPrefix(resource.Path);
                 fileEntry.Name = withoutPrefix;
                 _assetFiles.Remove(assetFilePath);
-                _assetFiles.Add(withoutPrefix, fileEntry);
+
```

Comment:
nit: Add a comment or use `dictionary[key] = value` to indicate this is an add/update and we always want the latest value.